index.js

```diff
@@ -0,0 +1,124 @@
+    // If this line is not present, calling sled.stdin.end with a chunk will
+    // result in an unhandled "Error: write EPIPE".
+    sled.stdin.on('error', () => {});
+
+    sled.stdin.end(JSON.stringify(input));
+  });
+}
+
+/**
+ * If a Cumulus Remote Message is passed, fetch it and return a full Cumulus Message
+ *
+ * @param {Object} cumulusMessage - either a full Cumulus Message or a Cumulus Remote Message
+ * @returns {Promise.<Object>} - a full Cumulus Message
+ */
+function loadRemoteEvent(cumulusMessage) {
+  return callCumulusMessageAdapter('loadRemoteEvent', { event: cumulusMessage });
+}
+
+/**
+ * Query AWS to create a task-specific event
+ *
+ * @param {Object} cumulusMessage - a full Cumulus Message
+ * @param {Object} context - an AWS Lambda context
+ * @returns {Promise.<Object>} - an Object containing the keys input, config and messageConfig
+ */
+function loadNestedEvent(cumulusMessage, context) {
+  return callCumulusMessageAdapter('loadNestedEvent', {
+    event: cumulusMessage,
+    context
+  });
+}
+
+/**
+ * Query AWS to create a task-specific event
+ *
+ * @param {Object} handlerResponse - the return value of the task function
+ * @param {Object} cumulusMessage - a full Cumulus Message
+ * @param {Object} messageConfig - the value of the messageConfig key returned by loadNestedEvent
+ * @returns {Promise.<Object>} - a Cumulus Message or a Cumulus Remote Message
+ */
+function createNextEvent(handlerResponse, cumulusMessage, messageConfig) {
+  return callCumulusMessageAdapter(
+    'createNextEvent',
+    {
+      event: cumulusMessage,
+      handler_response: handlerResponse,
+      message_config: messageConfig
+    }
+  );
+}
+
+/**
+ * Build a nested Cumulus event and pass it to a tasks's business function
+ *
+ * @param {Function} taskFunction - the function containing the business logic of the task
+ * @param {Object} cumulusMessage - either a full Cumulus Message or a Cumulus Remote Message
+ * @param {Object} context - an AWS Lambda context
+ * @param {Function} callback - the callback to be called when the taskFunction
+ *   has completed.  This should be the callback passed to the Lambda handler.
+ * @returns {Promise} - resolves when the task has completed
+ */
+function runCumulusTask(taskFunction, cumulusMessage, context, callback) {
+  const promisedRemoteEvent = loadRemoteEvent(cumulusMessage);
+  const promisedNestedEvent = promisedRemoteEvent.then((event) => loadNestedEvent(event, context));
+  const promisedTaskOutput = promisedNestedEvent
+    .then((nestedEvent) => taskFunction(nestedEvent, context));
+
+  return Promise.all([promisedTaskOutput, promisedRemoteEvent, promisedNestedEvent])
+    .then((resolvedPromises) =>
+      createNextEvent(resolvedPromises[0], resolvedPromises[1], resolvedPromises[2].messageConfig))
+    .then((nextEvent) => callback(null, nextEvent))
+    .catch((err) => {
+      if (err.name && err.name.includes('WorkflowError')) {
```

Comment:
I know this line is in the existing sled/message.js code in cumulus core repo but I'm wondering where the 'WorkflowError' error comes from - @yjpa7145 are you familiar with this error?

Reply:
I'm not.  My understanding is that, if an error is thrown by the task that's named `*WorkflowError*`, it will cause that error name to be included in the task output but the task will not itself fail.  I believe that something about the workflow will detect that and take appropriate action.  @bilts knows more about that.